format/private/format.sh

```diff
@@ -122,7 +119,22 @@ function ls-files {
           fi
           find_args+=("-name" "${patterns[$i]}")
         done
-        find "$@" "${find_args[@]}"
+        files=$(find "$@" "${find_args[@]}")
+    fi
+
+    if [[ $disable_git_attribute_checks == true ]]; then
+      echo $files
+      return
+    fi
+
+    if [[ $files != "" ]]; then
+        git_attributes=$(git check-attr rules-lint-ignored linguist-generated gitlab-generated -- $files)
```

Comment:
What I could see is, if some of the attributes is not specified, git will spit out things like `/path/to/file rules-lint-ignored: unspecified`, which means for each file we might have up to 3 lines of output, which I believe will in turn slow down grep.

`git check-attr -a -- /path/to/file` outputs nothing if no attribute is specified.